signer/tests/devenv/devenv.rs

```diff
@@ -0,0 +1,487 @@
+        height: 0,
+    };
+    let max_fee = deposit_amount / 2;
+    let (deposit_tx, deposit_request, _) = make_deposit_request(
+        &depositor,
+        deposit_amount,
+        depositor_utxo.clone(),
+        max_fee,
+        aggregate_key.into(),
+        recipient.clone().into(),
+    );
+    rpc.send_raw_transaction(&deposit_tx).unwrap();
+    let block_deposit_tx = faucet.generate_block();
+
+    let emily_request = CreateDepositRequestBody {
+        bitcoin_tx_output_index: deposit_request.outpoint.vout,
+        bitcoin_txid: deposit_request.outpoint.txid.to_string(),
+        deposit_script: deposit_request.deposit_script.to_hex_string(),
+        reclaim_script: deposit_request.reclaim_script.to_hex_string(),
+        transaction_hex: serialize_hex(&deposit_tx),
+    };
+
+    deposit_api::create_deposit(emily_client.config(), emily_request.clone())
+        .await
+        .expect("cannot create emily deposit");
+
+    // Wait for the signers to process the deposit
+    tokio::time::sleep(Duration::from_secs(10)).await;
+    faucet.generate_block();
+
+    // Then wait for the signers to mint the deposit
+    tokio::time::sleep(Duration::from_secs(10)).await;
+    faucet.generate_block();
+
+    // Check that the recipient did get the expected sBTC
+    let balance = get_sbtc_balance(&stacks, &deployer, &recipient.clone().into())
+        .await
+        .unwrap();
+    dbg!(&balance);
+    assert_ge!(balance.to_sat(), deposit_amount - max_fee);
+
+    // Fetch also the Stacks tx to check for its status after forking stacks.
+    let deposit_tx = &stacks_api_call::<serde_json::Value>("/extended/v2/addresses/SN3R84XZYA63QS28932XQF3G1J8R9PC3W76P9CSQS/transactions?limit=1&offset=0").await["results"][0]["tx"];
+
+    let deposit_tx_recipient = &deposit_tx["contract_call"]["function_args"][3]["repr"]
+        .as_str()
+        .unwrap()[1..];
+    let deposit_tx_status = deposit_tx["tx_status"].as_str().unwrap();
+    assert_eq!(deposit_tx_recipient, recipient.to_string());
+    assert_eq!(deposit_tx_status, "success");
+
+    let deposit_txid = deposit_tx["tx_id"].as_str().unwrap();
+
+    // Now we fork:
+    //  - we invalidate the deposit tx block via rpc
+    rpc.invalidate_block(&block_deposit_tx).unwrap();
+
+    //  - we invalidate the deposit tx by spending its vin
+    let invlidating_deposit_tx = create_nop_transaction(
+        &depositor,
+        depositor_utxo.clone(),
+        // must be more than the package (deposit tx + sweep) in mempool to be
+        // accepted as RBF
+        Amount::from_sat(deposit_amount - 1000),
+    )
+    .await;
+    let invlidating_deposit_txid = rpc.send_raw_transaction(&invlidating_deposit_tx).unwrap();
+
+    rpc.call::<GenerateBlockJson>(
+        "generateblock",
+        &[
+            faucet.address.to_string().into(),
+            to_value(&[invlidating_deposit_txid.to_string()]).unwrap(),
+        ],
+    )
+    .unwrap();
```

Comment:
Ahhh, this ensures that the invalidating transaction gets confirmed. I'd add a comment to that effect here.